src/Uno.Toolkit.UI/Controls/DrawerControl/DrawerControl.cs

```diff
@@ -261,39 +264,43 @@ private void OnManipulationCompleted(object sender, ManipulationCompletedRoutedE
 			var isInCorrectDirection = Math.Sign(cumulative) == (IsOpen ^ UseNegativeTranslation() ? 1 : -1);
 			var isPastThresholdRatio = Math.Abs(cumulative / length) >= DragToggleThresholdRatio;
 
-			UpdateIsOpen(IsOpen ^ (isInCorrectDirection && isPastThresholdRatio));
+			UpdateIsOpen(IsOpen ^ (isInCorrectDirection && isPastThresholdRatio), shouldAnimate: true);
 		}
 
 		private void OnSizeChanged(object sender, SizeChangedEventArgs e)
 		{
 			UpdateSwipeContentPresenterSize();
-			UpdateIsOpen(IsOpen, animate: false);
+			UpdateIsOpen(IsOpen, shouldAnimate: false);
 		}
 
 		private void OnLightDismissOverlayTapped(object sender, TappedRoutedEventArgs e)
 		{
 			if (!IsLightDismissEnabled) return;
 
 			StopRunningAnimation();
-			UpdateIsOpen(false);
+			UpdateIsOpen(false, shouldAnimate: true);
 		}
 
-		private void UpdateIsOpen(bool willBeOpen, bool animate = true)
+		private void UpdateIsOpen(bool willBeOpen, bool shouldAnimate = true)
 		{
 			var length = GetActualDrawerDepth();
 			var currentOffset = TranslateOffset;
 			var targetOffset = GetSnappingOffsetFor(willBeOpen);
 			var relativeDistanceRatio = Math.Abs(Math.Abs(currentOffset) - Math.Abs(targetOffset)) / length;
 
-			if (!animate || ((1 - relativeDistanceRatio) >= AnimateSnappingThresholdRatio))
+			if (shouldAnimate &&
+				IsLoaded &&
+				_dispatcher.HasThreadAccess &&
+				length > 0 && // skip animation if we have nothing to animate (either from not being ready, or no valid content)
+				!((1 - relativeDistanceRatio) < AnimateSnappingThresholdRatio)) // skip animation if we are 5% from done
```

Comment:
Why negate a condition? I've the feeling that the previous condition was clearer.

```suggestion
				((1 - relativeDistanceRatio) >= AnimateSnappingThresholdRatio)) // skip animation if we are 5% from done
```